server/src/main/java/org/opensearch/common/util/concurrent/OpenSearchAbortPolicy.java

```diff
@@ -64,6 +68,10 @@ public void rejectedExecution(Runnable r, ThreadPoolExecutor executor) {
             }
         }
         rejected.inc();
+        if (executor.isTerminating() || executor.isTerminated() || executor.isShutdown()) {
```

Comment:
@peternied we should not that: the listeners will be called on rejection to notify the caller that the submission failed, with this change it won't happen anymore, leaving some callbacks in dangling state

Reply:
> I understand from the programming purity - this is against the conventions of executors; however, aren't the only cases where we are shutting down executors when we are shutting down nodes?

It may be the case, but even than we need a clean shutdown, tracing comes immediately as an example here - shutting down the node should gracefully flush the data to collector but with this change, the  trace spans won't be flushed since they will be left in dangling state.

Reply:
To revisit;
> Is there a better way to determine that the cluster is going down so these sources of failures can be ignored?

I’ve got two ways forward that I’ll pursue:

1) The stale state detection task in seg-rep is not-useful while a node is shutting down maybe I can dequeue or swallow the scheduling from that system. 

2) Failing that, I’m going to see if there is a way to keep honoring the contract - if there are callbacks - but not in other scenarios.

Reply:
> The stale state detection task in seg-rep is not-useful while a node is shutting down maybe I can dequeue or swallow the scheduling from that system.

In some places, there are lifecycle check, when node goes down, the lifecycle check may not even proceed with the tasks (I would strongly advocate to not alter the pull behaviour - this is low level tool).

Reply:
I believe I found a way forward using the first path which is much cleaner in my mind, not 100% sure how I can validate other than get a number of test runs in.  Let me know what you think of this.